ext/lib/ipc/libmetal.cmake

```diff
@@ -0,0 +1,12 @@
+include(ExternalProject)
+
+ExternalProject_Add(
+  libmetal                 # Name for custom target
+  SOURCE_DIR $ENV{ZEPHYR_BASE}/ext/lib/ipc/libmetal/
+  INSTALL_COMMAND ""      # This particular build system has no install command
+  CMAKE_ARGS -DWITH_ZEPHYR=ON -DBOARD=${BOARD} -DWITH_DEFAULT_LOGGER=OFF -DWITH_DOC=OFF
```

Comment:
Recursive-make must be avoided when possible.

For OpenThread it was not possible due to the sheer size of the OpenThread build system.

For libmetal it is possible to avoid recursive-make because the project is small and easy to build.

Reply:
> Recursive-make must be avoided when possible.
> 
> For OpenThread it was not possible due to the sheer size of the OpenThread build system.
> 
> For libmetal it is possible to avoid recursive-make because the project is small and easy to build.

I'm not going to address this for 1.12.  We will fix this for 1.13 when we import the new openamp code.